### PR TITLE
SmartContract: fix native NEP-17 manifest construction

### LINKED ISSUE #3194: HF-based native contract update doesn't work as expected
**Describe the bug**
While testing the [NeoGo's port](https://github.com/nspcc-dev/neo-go/pull/3402) of [HF-based native contracts update](#2942) I discovered the following situation: native NEP-17 tokens do not have `Transfer` events declared in their manifest. So states comparison of the fresh NeoC# node master with https://github.com/nspcc-dev/neo-go/pull/3402 gives us:
```
anna@kiwi:~/Documents/GitProjects/nspcc-dev/neo-go$ go run scripts/compare-states/compare-states.go --ignore-height http://localhost:10332 http://localhost:20332
at 0: e96672c15146f6e385c87f4d8327207e33086f805e88b4a28fb8cffe65f82786 vs 0dcd4ee76db1f5e30eef0c1574311063adcea654a22831430ef8e5a3f0e17eba
state differs at 0, block 1f4d1defa46faa5e7b9b8d3f79a06bec777d7c26c4aa5f6f5899a291daa87c15
different state found
exit status 1
```
And if take a look at the genesis block state dumps we can see that:
```
anna@kiwi:~/Documents/GitProjects/nspcc-dev/neo-go$ go run scripts/compare-dumps/compare-dumps.go Storage_mainnet_783K/ GoDump_mainnet_10/
Processing directory BlockStorage_0
block 0: value mismatch for key /////wj1Y+pAvCg9TQ4FxI6jBbPyoHNA7w==:
	Contract ID: -1
	Item prefix (base64): CPVj6kC8KD1NDgXEjqMFs/Kgc0Dv
	Item prefix (hex): 08f563ea40bc283d4d0e05c48ea305b3f2a07340ef
	Item prefix (bytes): [8 245 99 234 64 188 40 61 77 14 5 196 142 163 5 179 242 160 115 64 239]
	Diff: QAUhAfshACgU9WPqQLwoPU0OBcSOowWz8qBzQO8o005FRjNuZW8tY29yZS12My4wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACFEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQHvz5gNBCCgITmVvVG9rZW5AAEgAQAEoBk5FUC0xN0ECQBNBBSgJYmFsYW5jZU9mQAFBAigHYWNjb3VudCEBFCEBESEAIAFBBSgIZGVjaW1hbHNAACEBESEBByABQQUoD2dldEFjY291bnRTdGF0ZUABQQIoB2FjY291bnQhARQhASAhAQ4gAUEFKBBnZXRBbGxDYW5kaWRhdGVzQAAhATAhARUgAUEFKBBnZXRDYW5kaWRhdGVWb3RlQAFBAigGcHViS2V5IQEWIQERIQEcIAFBBSgNZ2V0Q2FuZGlkYXRlc0AAIQEgIQEjIAFBBSgMZ2V0Q29tbWl0dGVlQAAhASAhASogAUEFKA5nZXRHYXNQZXJCbG9ja0AAIQERIQExIAFBBSgWZ2V0TmV4dEJsb2NrVmFsaWRhdG9yc0AAIQEgIQE4IAFBBSgQZ2V0UmVnaXN0ZXJQcmljZUAAIQERIQE/IAFBBSgRcmVnaXN0ZXJDYW5kaWRhdGVAAUECKAZwdWJrZXkhARYhARAhAUYgAEEFKA5zZXRHYXNQZXJCbG9ja0ABQQIoC2dhc1BlckJsb2NrIQERIQL/ACEBTSAAQQUoEHNldFJlZ2lzdGVyUHJpY2VAAUECKA1yZWdpc3RlclByaWNlIQERIQL/ACEBVCAAQQUoBnN5bWJvbEAAIQETIQFbIAFBBSgLdG90YWxTdXBwbHlAACEBESEBYiABQQUoCHRyYW5zZmVyQARBAigEZnJvbSEBFEECKAJ0byEBFEECKAZhbW91bnQhARFBAigEZGF0YSEAIQEQIQFpIABBBSgMdW5jbGFpbWVkR2FzQAJBAigHYWNjb3VudCEBFEECKANlbmQhAREhAREhAXAgAUEFKBN1bnJlZ2lzdGVyQ2FuZGlkYXRlQAFBAigGcHVia2V5IQEWIQEQIQF3IABBBSgEdm90ZUACQQIoB2FjY291bnQhARRBAigGdm90ZVRvIQEWIQEQIQF+IABAA0ECKBVDYW5kaWRhdGVTdGF0ZUNoYW5nZWRAA0ECKAZwdWJrZXkhARZBAigKcmVnaXN0ZXJlZCEBEEECKAV2b3RlcyEBEUECKARWb3RlQARBAigHYWNjb3VudCEBFEECKARmcm9tIQEWQQIoAnRvIQEWQQIoBmFtb3VudCEBEUECKBBDb21taXR0ZWVDaGFuZ2VkQAJBAigDb2xkIQEgQQIoA25ldyEBIEABQQIAAEAAKARudWxs vs QAUhAfshACgU9WPqQLwoPU0OBcSOowWz8qBzQO8o005FRjNuZW8tY29yZS12My4wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACFEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQHvz5gNBCCgITmVvVG9rZW5AAEgAQAEoBk5FUC0xN0ECQBNBBSgJYmFsYW5jZU9mQAFBAigHYWNjb3VudCEBFCEBESEAIAFBBSgIZGVjaW1hbHNAACEBESEBByABQQUoD2dldEFjY291bnRTdGF0ZUABQQIoB2FjY291bnQhARQhASAhAQ4gAUEFKBBnZXRBbGxDYW5kaWRhdGVzQAAhATAhARUgAUEFKBBnZXRDYW5kaWRhdGVWb3RlQAFBAigGcHViS2V5IQEWIQERIQEcIAFBBSgNZ2V0Q2FuZGlkYXRlc0AAIQEgIQEjIAFBBSgMZ2V0Q29tbWl0dGVlQAAhASAhASogAUEFKA5nZXRHYXNQZXJCbG9ja0AAIQERIQExIAFBBSgWZ2V0TmV4dEJsb2NrVmFsaWRhdG9yc0AAIQEgIQE4IAFBBSgQZ2V0UmVnaXN0ZXJQcmljZUAAIQERIQE/IAFBBSgRcmVnaXN0ZXJDYW5kaWRhdGVAAUECKAZwdWJrZXkhARYhARAhAUYgAEEFKA5zZXRHYXNQZXJCbG9ja0ABQQIoC2dhc1BlckJsb2NrIQERIQL/ACEBTSAAQQUoEHNldFJlZ2lzdGVyUHJpY2VAAUECKA1yZWdpc3RlclByaWNlIQERIQL/ACEBVCAAQQUoBnN5bWJvbEAAIQETIQFbIAFBBSgLdG90YWxTdXBwbHlAACEBESEBYiABQQUoCHRyYW5zZmVyQARBAigEZnJvbSEBFEECKAJ0byEBFEECKAZhbW91bnQhARFBAigEZGF0YSEAIQEQIQFpIABBBSgMdW5jbGFpbWVkR2FzQAJBAigHYWNjb3VudCEBFEECKANlbmQhAREhAREhAXAgAUEFKBN1bnJlZ2lzdGVyQ2FuZGlkYXRlQAFBAigGcHVia2V5IQEWIQEQIQF3IABBBSgEdm90ZUACQQIoB2FjY291bnQhARRBAigGdm90ZVRvIQEWIQEQIQF+IABABEECKAhUcmFuc2ZlckADQQIoBGZyb20hARRBAigCdG8hARRBAigGYW1vdW50IQERQQIoFUNhbmRpZGF0ZVN0YXRlQ2hhbmdlZEADQQIoBnB1YmtleSEBFkECKApyZWdpc3RlcmVkIQEQQQIoBXZvdGVzIQERQQIoBFZvdGVABEECKAdhY2NvdW50IQEUQQIoBGZyb20hARZBAigCdG8hARZBAigGYW1vdW50IQERQQIoEENvbW1pdHRlZUNoYW5nZWRAAkECKANvbGQhASBBAigDbmV3IQEgQAFBAgAAQAAoBG51bGw=
block 0: value mismatch for key /////wjPduKL0AYsSkeO41VhARMZ88+k0g==:
	Contract ID: -1
	Item prefix (base64): CM924ovQBixKR47jVWEBExnzz6TS
	Item prefix (hex): 08cf76e28bd0062c4a478ee35561011319f3cfa4d2
	Item prefix (bytes): [8 207 118 226 139 208 6 44 74 71 142 227 85 97 1 19 25 243 207 164 210]
	Diff: QAUhAfohACgUz3bii9AGLEpHjuNVYQETGfPPpNIocU5FRjNuZW8tY29yZS12My4wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAjEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0BRQceeQQgoCEdhc1Rva2VuQABIAEABKAZORVAtMTdBAkAFQQUoCWJhbGFuY2VPZkABQQIoB2FjY291bnQhARQhAREhACABQQUoCGRlY2ltYWxzQAAhAREhAQcgAUEFKAZzeW1ib2xAACEBEyEBDiABQQUoC3RvdGFsU3VwcGx5QAAhAREhARUgAUEFKAh0cmFuc2ZlckAEQQIoBGZyb20hARRBAigCdG8hARRBAigGYW1vdW50IQERQQIoBGRhdGEhACEBECEBHCAAQABAAUECAABAACgEbnVsbA== vs QAUhAfohACgUz3bii9AGLEpHjuNVYQETGfPPpNIocU5FRjNuZW8tY29yZS12My4wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAjEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0BRQceeQQgoCEdhc1Rva2VuQABIAEABKAZORVAtMTdBAkAFQQUoCWJhbGFuY2VPZkABQQIoB2FjY291bnQhARQhAREhACABQQUoCGRlY2ltYWxzQAAhAREhAQcgAUEFKAZzeW1ib2xAACEBEyEBDiABQQUoC3RvdGFsU3VwcGx5QAAhAREhARUgAUEFKAh0cmFuc2ZlckAEQQIoBGZyb20hARRBAigCdG8hARRBAigGYW1vdW50IQERQQIoBGRhdGEhACEBECEBHCAAQAFBAigIVHJhbnNmZXJAA0ECKARmcm9tIQEUQQIoAnRvIQEUQQIoBmFtb3VudCEBEUABQQIAAEAAKARudWxs
file BlockStorage_0/dump-block-0.json: fail
compare-dumps dumpDirA dumpDirB
exit status 1
```

So there are two state differences in genesis block. Contract `-1` is native Management, it stores contract states by `0x08` prefix, `f563ea40bc283d4d0e05c48ea305b3f2a07340ef` and `cf76e28bd0062c4a478ee35561011319f3cfa4d2` are Neo and Gas token contracts hashes correspondingly, and the resulting contract states differ from NeoGo at declared `events` field:
image

Also, `getcontractstate` call to NeoC# RPC node gives us the following result (`Transfer` events are missing in the declaration):
```
anna@kiwi:~/Documents/GitProjects/nspcc-dev/neo-go$ curl -d '{ "jsonrpc": "2.0", "id": 1, "method": "getcontractstate", "params": ["0xd2a4cff31913016155e38e474a2c06d08be276cf"] }' localhost:10332 | json_pp
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1159    0  1042  100   117  11336   1272 --:--:-- --:--:-- --:--:-- 12736
{
   "id" : 1,
   "jsonrpc" : "2.0",
   "result" : {
      "hash" : "0xd2a4cff31913016155e38e474a2c06d08be276cf",
      "id" : -6,
      "manifest" : {
         "abi" : {
            "events" : [],
            "methods" : [
               {
                  "name" : "balanceOf",
                  "offset" : 0,
                  "parameters" : [
                     {
                        "name" : "account",
                        "type" : "Hash160"
                     }
                  ],
                  "returntype" : "Integer",
                  "safe" : true
               },
               {
                  "name" : "decimals",
                  "offset" : 7,
                  "parameters" : [],
                  "returntype" : "Integer",
                  "safe" : true
               },
               {
                  "name" : "symbol",
                  "offset" : 14,
                  "parameters" : [],
                  "returntype" : "String",
                  "safe" : true
               },
               {
                  "name" : "totalSupply",
                  "offset" : 21,
                  "parameters" : [],
                  "returntype" : "Integer",
                  "safe" : true
               },
               {
                  "name" : "transfer",
                  "offset" : 28,
                  "parameters" : [
                     {
                        "name" : "from",
                        "type" : "Hash160"
                     },
                     {
                        "name" : "to",
                        "type" : "Hash160"
                     },
                     {
                        "name" : "amount",
                        "type" : "Integer"
                     },
                     {
                        "name" : "data",
                        "type" : "Any"
                     }
                  ],
                  "returntype" : "Boolean",
                  "safe" : false
               }
            ]
         },
         "extra" : null,
         "features" : {},
         "groups" : [],
         "name" : "GasToken",
         "permissions" : [
            {
               "contract" : "*",
               "methods" : "*"
            }
         ],
         "supportedstandards" : [
            "NEP-17"
         ],
         "trusts" : []
      },
      "nef" : {
         "checksum" : 2663858513,
         "compiler" : "neo-core-v3.0",
         "magic" : 860243278,
         "script" : "EEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0A=",
         "source" : "",
         "tokens" : []
      },
      "updatecounter" : 0
   }
}
```

**To Reproduce**
1. Start NeoC# node with RpcServer plugin and StatesDumper plugin;
2. Check the state dumps of genesis block;
3. Check the result of `getcontractstate` RPC call for Neo or Gas native contracts.

**Expected behavior**
Both Neo and Gas should have proper events declared in the manifest.

**Platform:**
 - Version: 3a12f3dd4d1400e842d58a12f0057607627e511e

**(Optional) Additional context**
This bug is critical for 3.7, we should fix it before the release.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
# Description

Take into account event attributes declared onto the base class while constructing manifest for native contracts. Close #3194.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [X] `getcontractstate` RPC call for native NeoToken and GasToken sent to the updated node. The result is expected and correct, see the result:
<details>
  <summary>Updated NeoToken contract state</summary>
 
```
anna@kiwi:~/Documents/GitProjects/nspcc-dev/neo-go$ curl -d '{ "jsonrpc": "2.0", "id": 1, "method": "getcontractstate", "params": [-5] }' localhost:10332 | json_pp
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3433    0  3358  100    75  1287k  29446 --:--:-- --:--:-- --:--:-- 1676k
{
   "id" : 1,
   "jsonrpc" : "2.0",
   "result" : {
      "hash" : "0xef4073a0f2b305a38ec4050e4d3d28bc40ea63f5",
      "id" : -5,
      "manifest" : {
         "abi" : {
            "events" : [
               {
                  "name" : "Transfer",
                  "parameters" : [
                     {
                        "name" : "from",
                        "type" : "Hash160"
                     },
                     {
                        "name" : "to",
                        "type" : "Hash160"
                     },
                     {
                        "name" : "amount",
                        "type" : "Integer"
                     }
                  ]
               },
               {
                  "name" : "CandidateStateChanged",
                  "parameters" : [
                     {
                        "name" : "pubkey",
                        "type" : "PublicKey"
                     },
                     {
                        "name" : "registered",
                        "type" : "Boolean"
                     },
                     {
                        "name" : "votes",
                        "type" : "Integer"
                     }
                  ]
               },
               {
                  "name" : "Vote",
                  "parameters" : [
                     {
                        "name" : "account",
                        "type" : "Hash160"
                     },
                     {
                        "name" : "from",
                        "type" : "PublicKey"
                     },
                     {
                        "name" : "to",
                        "type" : "PublicKey"
                     },
                     {
                        "name" : "amount",
                        "type" : "Integer"
                     }
                  ]
               },
               {
                  "name" : "CommitteeChanged",
                  "parameters" : [
                     {
                        "name" : "old",
                        "type" : "Array"
                     },
                     {
                        "name" : "new",
                        "type" : "Array"
                     }
                  ]
               }
            ],
            "methods" : [
               {
                  "name" : "balanceOf",
                  "offset" : 0,
                  "parameters" : [
                     {
                        "name" : "account",
                        "type" : "Hash160"
                     }
                  ],
                  "returntype" : "Integer",
                  "safe" : true
               },
               {
                  "name" : "decimals",
                  "offset" : 7,
                  "parameters" : [],
                  "returntype" : "Integer",
                  "safe" : true
               },
               {
                  "name" : "getAccountState",
                  "offset" : 14,
                  "parameters" : [
                     {
                        "name" : "account",
                        "type" : "Hash160"
                     }
                  ],
                  "returntype" : "Array",
                  "safe" : true
               },
               {
                  "name" : "getAllCandidates",
                  "offset" : 21,
                  "parameters" : [],
                  "returntype" : "InteropInterface",
                  "safe" : true
               },
               {
                  "name" : "getCandidateVote",
                  "offset" : 28,
                  "parameters" : [
                     {
                        "name" : "pubKey",
                        "type" : "PublicKey"
                     }
                  ],
                  "returntype" : "Integer",
                  "safe" : true
               },
               {
                  "name" : "getCandidates",
                  "offset" : 35,
                  "parameters" : [],
                  "returntype" : "Array",
                  "safe" : true
               },
               {
                  "name" : "getCommittee",
                  "offset" : 42,
                  "parameters" : [],
                  "returntype" : "Array",
                  "safe" : true
               },
               {
                  "name" : "getGasPerBlock",
                  "offset" : 49,
                  "parameters" : [],
                  "returntype" : "Integer",
                  "safe" : true
               },
               {
                  "name" : "getNextBlockValidators",
                  "offset" : 56,
                  "parameters" : [],
                  "returntype" : "Array",
                  "safe" : true
               },
               {
                  "name" : "getRegisterPrice",
                  "offset" : 63,
                  "parameters" : [],
                  "returntype" : "Integer",
                  "safe" : true
               },
               {
                  "name" : "registerCandidate",
                  "offset" : 70,
                  "parameters" : [
                     {
                        "name" : "pubkey",
                        "type" : "PublicKey"
                     }
                  ],
                  "returntype" : "Boolean",
                  "safe" : false
               },
               {
                  "name" : "setGasPerBlock",
                  "offset" : 77,
                  "parameters" : [
                     {
                        "name" : "gasPerBlock",
                        "type" : "Integer"
                     }
                  ],
                  "returntype" : "Void",
                  "safe" : false
               },
               {
                  "name" : "setRegisterPrice",
                  "offset" : 84,
                  "parameters" : [
                     {
                        "name" : "registerPrice",
                        "type" : "Integer"
                     }
                  ],
                  "returntype" : "Void",
                  "safe" : false
               },
               {
                  "name" : "symbol",
                  "offset" : 91,
                  "parameters" : [],
                  "returntype" : "String",
                  "safe" : true
               },
               {
                  "name" : "totalSupply",
                  "offset" : 98,
                  "parameters" : [],
                  "returntype" : "Integer",
                  "safe" : true
               },
               {
                  "name" : "transfer",
                  "offset" : 105,
                  "parameters" : [
                     {
                        "name" : "from",
                        "type" : "Hash160"
                     },
                     {
                        "name" : "to",
                        "type" : "Hash160"
                     },
                     {
                        "name" : "amount",
                        "type" : "Integer"
                     },
                     {
                        "name" : "data",
                        "type" : "Any"
                     }
                  ],
                  "returntype" : "Boolean",
                  "safe" : false
               },
               {
                  "name" : "unclaimedGas",
                  "offset" : 112,
                  "parameters" : [
                     {
                        "name" : "account",
                        "type" : "Hash160"
                     },
                     {
                        "name" : "end",
                        "type" : "Integer"
                     }
                  ],
                  "returntype" : "Integer",
                  "safe" : true
               },
               {
                  "name" : "unregisterCandidate",
                  "offset" : 119,
                  "parameters" : [
                     {
                        "name" : "pubkey",
                        "type" : "PublicKey"
                     }
                  ],
                  "returntype" : "Boolean",
                  "safe" : false
               },
               {
                  "name" : "vote",
                  "offset" : 126,
                  "parameters" : [
                     {
                        "name" : "account",
                        "type" : "Hash160"
                     },
                     {
                        "name" : "voteTo",
                        "type" : "PublicKey"
                     }
                  ],
                  "returntype" : "Boolean",
                  "safe" : false
               }
            ]
         },
         "extra" : null,
         "features" : {},
         "groups" : [],
         "name" : "NeoToken",
         "permissions" : [
            {
               "contract" : "*",
               "methods" : "*"
            }
         ],
         "supportedstandards" : [
            "NEP-17"
         ],
         "trusts" : []
      },
      "nef" : {
         "checksum" : 65467259,
         "compiler" : "neo-core-v3.0",
         "magic" : 860243278,
         "script" : "EEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQA==",
         "source" : "",
         "tokens" : []
      },
      "updatecounter" : 0
   }
}
```

</details>

<details>
  <summary>Updated GasToken contract state</summary>
 
```
anna@kiwi:~/Documents/GitProjects/nspcc-dev/neo-go$ curl -d '{ "jsonrpc": "2.0", "id": 1, "method": "getcontractstate", "params": [-6] }' localhost:10332 | json_pp
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1250    0  1175  100    75   408k  26709 --:--:-- --:--:-- --:--:--  610k
{
   "id" : 1,
   "jsonrpc" : "2.0",
   "result" : {
      "hash" : "0xd2a4cff31913016155e38e474a2c06d08be276cf",
      "id" : -6,
      "manifest" : {
         "abi" : {
            "events" : [
               {
                  "name" : "Transfer",
                  "parameters" : [
                     {
                        "name" : "from",
                        "type" : "Hash160"
                     },
                     {
                        "name" : "to",
                        "type" : "Hash160"
                     },
                     {
                        "name" : "amount",
                        "type" : "Integer"
                     }
                  ]
               }
            ],
            "methods" : [
               {
                  "name" : "balanceOf",
                  "offset" : 0,
                  "parameters" : [
                     {
                        "name" : "account",
                        "type" : "Hash160"
                     }
                  ],
                  "returntype" : "Integer",
                  "safe" : true
               },
               {
                  "name" : "decimals",
                  "offset" : 7,
                  "parameters" : [],
                  "returntype" : "Integer",
                  "safe" : true
               },
               {
                  "name" : "symbol",
                  "offset" : 14,
                  "parameters" : [],
                  "returntype" : "String",
                  "safe" : true
               },
               {
                  "name" : "totalSupply",
                  "offset" : 21,
                  "parameters" : [],
                  "returntype" : "Integer",
                  "safe" : true
               },
               {
                  "name" : "transfer",
                  "offset" : 28,
                  "parameters" : [
                     {
                        "name" : "from",
                        "type" : "Hash160"
                     },
                     {
                        "name" : "to",
                        "type" : "Hash160"
                     },
                     {
                        "name" : "amount",
                        "type" : "Integer"
                     },
                     {
                        "name" : "data",
                        "type" : "Any"
                     }
                  ],
                  "returntype" : "Boolean",
                  "safe" : false
               }
            ]
         },
         "extra" : null,
         "features" : {},
         "groups" : [],
         "name" : "GasToken",
         "permissions" : [
            {
               "contract" : "*",
               "methods" : "*"
            }
         ],
         "supportedstandards" : [
            "NEP-17"
         ],
         "trusts" : []
      },
      "nef" : {
         "checksum" : 2663858513,
         "compiler" : "neo-core-v3.0",
         "magic" : 860243278,
         "script" : "EEEa93tnQBBBGvd7Z0AQQRr3e2dAEEEa93tnQBBBGvd7Z0A=",
         "source" : "",
         "tokens" : []
      },
      "updatecounter" : 0
   }
}
```

</details>

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by roman-khimov — DISMISSED** [review]
I don't want to think about the case with overrides of some kind for the same event name. Hopefully, we'll never do this for natives.

**Comment by dusmart** [comment]
Thanks for locating this bug.
BTW, this fix is also a hard fork.
Please take care when performing next update @superboyiii. We should make sure no user has used this bug.

**Comment by AnnaShaleva** [comment]
> BTW, this fix is also a hard fork.

It’s not. This bug was introduced in post-3.6, in #2942, so this code is not in production yet, and the fix itself just fixes natives manifest and Management’s storage.

But you’re right from the point that this fix requires node resync from genesis to work.

**Comment by Jim8y** [comment]
Any possible UT for this fix?

**Review by Jim8y — DISMISSED** [review]

**Comment by AnnaShaleva** [comment]
> Any possible UT for this fix?

Sure, will add in a couple of hours.

**Review by shargon — DISMISSED** [review]

**Review by vncoelho — COMMENTED** [review]
Isn't it possible to set up  a UT to ensure that this fix is correct?

**Comment by vncoelho** [comment]
> Any possible UT for this fix?

I just saw @Jim8y request now. I was checking the code.

The code looks good and makes sense. It would be good to ensure a failed UT now and the fix for it.

**Comment by vncoelho** [comment]
This reminds me that it would be also good to generate automatically manifest for NEO and GAS instead of manually. That would had make this error appear earlier perhaps.

**Comment by Jim8y** [comment]
> This reminds me that it would be also good to generate automatically manifest for NEO and GAS instead of manually. That would had make this error appear earlier perhaps.

Totally agree.

**Comment by AnnaShaleva** [comment]
> Sure, will add in a couple of hours.

Two tests are added: one for particularly this bug (check `Transfer` events for NEP17 native contracts), and another one to check the whole set of native contract states at genesis block with all hardforks disabled. The expected manifests are checked precisely, so they are correct. Also checked against NeoGo's ones.

In future `TestGenesisNativeState` will help us to detect similar bugs.

**Review by Jim8y — APPROVED** [review]

**Comment by AnnaShaleva** [comment]
>  at genesis block with all hardforks disabled

With all hardforks enabled, actually. Let me fix this.

**Comment by AnnaShaleva** [comment]
> With all hardforks enabled, actually. Let me fix this.

Fixed.

**Review by vncoelho — DISMISSED** [review]

**Review by Jim8y — APPROVED** [review]

**Comment by Jim8y** [comment]
> > With all hardforks enabled, actually. Let me fix this.
> 
> Fixed.

wait a sec, what is the new logic? hardforks enabled or disabled?

**Comment by AnnaShaleva** [comment]
> wait a sec, what is the new logic? hardforks enabled or disabled?

The code itself preserves the old logic except that NEP17 notifications are included into the NEP17 state, so #3194 is fixed.

The unit tests perform the presence of these notifications and check natives state with all hardforks *enabled* since it's the default behaviour of `TestBlockchain` (it uses the default `ProtocolSettings`). I think it's perfectly OK for our tests, since we need to test fresh functionality as far. But for this particular bug it doesn't make any difference since `Transfer` notifications should always be present in the native state (with or without hardforks).

**Review by shargon — APPROVED** [review]

**Review by roman-khimov — APPROVED** [review]

File: tests/Neo.UnitTests/SmartContract/Native/UT_NativeContract.cs [hunk 1]
```diff
@@ -43,5 +83,77 @@ public void TestIsInitializeBlock()
...
+                    NextConsensus = UInt160.Zero,
+                    PrevHash = UInt256.Zero,
+                    Witness = new Witness() { InvocationScript = Array.Empty<byte>(), VerificationScript = Array.Empty<byte>() }
+                },
+                Transactions = []
+            };
+            var snapshot = _snapshot.CreateSnapshot();
+
+            // Ensure that native NEP17 contracts contain proper supported standards and events declared
+            // in the manifest constructed for all hardforks enabled. Ref. #3195.
```
Comment (Jim8y) [inline tests/Neo.UnitTests/SmartContract/Native/UT_NativeContract.cs:105]:
```suggestion
            // in the manifest constructed for all hardforks enabled. Ref. https://github.com/neo-project/neo/pull/3195.
```

**Comment by Jim8y** [comment]
@shargon can be merged.

**Review by Jim8y — APPROVED** [review]

**Review by superboyiii — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Pull request merged**